src/viewer.js

```diff
@@ -195,6 +203,7 @@ export class Viewer {
 
     // Wait for document to become visible.
     this.docState_.onVisibilityChanged(() => {
+      this.onVisibilityResolve_();
```

Comment:
Two issues here:

1. The promise might already need to be resolved automatically by the end of the `Viewer` constructor if the document is immediately visible.
2. When handling the visibility event, you have to check `Viewer.isVisible()` before resolving this promise (notice that this method already checks for `!docstate.isHidden`). I think you just need to put this code in a private method in Viewer, e.g. "handleVisibility_" where you can fire `visibilityObservable_`  and resolve the visibility promise.

But keep in mind, again, that you may need to resolve the promise in the constructor when it's already visible.